### ISSUE TITLE
Default HeliographicStonyhurst __repr__ is hard to parse by eye

### ISSUE THREAD

**Issue by dstansby** [sunpy/sunpy#5662]
![Screenshot 2021-11-02 at 11 02 45](https://user-images.githubusercontent.com/6197628/139835102-ba960b09-8f9b-471e-8aca-2c2ab24bf9e6.png)

Personally I find it quite hard to parse this by eye and pick out the useful information. Some thoughts as to why this might be:
- The actual coordinate values are all the way down on the last line
- The observer coordinate values are on the same line as the  for the coordinate units
- There's a line break between observer coordinate units and observer coordinate units

This is probably something that needs to be discussed and fixed upstream, but reporting here for now.

**Comment by Cadair** [comment]
> The actual coordinate values are all the way down on the last line

When the attributes on the class are less insane this actually makes a lot of sense. If you look at the repr for an `observer='earth'` coordinate it's clearer to see why this is like this.

**Comment by ayshih** [comment]
Proof of concept for discussion:
```
<Helioprojective Coordinate (obstime=2011-06-07T06:33:02.770,
                             rsun=696000.0 km,
                             observer=<HeliographicStonyhurst Coordinate (obstime=2011-06-07T06:33:02.770,
                                                                          rsun=696000.0 km):
                                      (lon, lat, radius) in (deg, deg, m)
                                          (-0.00406308, 0.04787238, 1.51846026e+11)>):
(Tx, Ty) in arcsec
    (3.22309951, 1.38578135)>
```
It's more readable, but takes a lot more space.

**Comment by Cadair** [comment]
What would be nice (but probably hard) is hiding the attrs of the observer if they match the parent frame?

**Comment by ayshih** [comment]
It'd be doable in this case, but a general implementation could be tricky.  As an extreme example, imagine that you have a coordinate frame (A) that has two coordinate frame attributes (B and C), and each of those each also has a coordinate frame attribute (D and E, respectively).  If the obstimes are:
```
A (obstime=t1)
    B (obstime=t2)
        D (obstime=t2)
    C (obstime=t1)
        E (obstime=t2)
```
would this be confusing?
```
A (obstime=t1)
    B (obstime=t2)
        D (obstime=...)
    C (obstime=...)
        E (obstime=t2)
```

**Comment by dstansby** [comment]
Just revisiting this, and I think going from this (current):
```
<SkyCoord (HeliographicCarrington: obstime=2020-04-03T13:34:38.000, rsun=696000.0 km, observer=<HeliographicStonyhurst Coordinate (obstime=2020-04-03T13:34:38.000, rsun=696000.0 km): (lon, lat, radius) in (deg, deg, m)
    (0.08878735, -6.37561893, 1.49588478e+11)>): 
    (258.57809448, 28.85744858)>
```
to putting the units *after* the values:
```
<SkyCoord (HeliographicCarrington: obstime=2020-04-03T13:34:38.000, rsun=696000.0 km, observer=<HeliographicStonyhurst Coordinate (obstime=2020-04-03T13:34:38.000, rsun=696000.0 km): 
    (0.08878735deg, -6.37561893, 1.49588478e+11) (lon, lat, radius) in (deg, deg, m)>): 
    (258.57809448, 28.85744858) (lon, lat) in deg>
```
is much more readable. I think making a change like this would be an upstream `astropy` issue though?
